api/app/validators/SlotValidator.scala

```diff
@@ -0,0 +1,26 @@
+package validators
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import models.Slot
+import persistence.Store
+
+sealed trait Validation
+final case object Success extends Validation
+final case class Failure(message: String) extends Validation
+
+case class SlotValidator(store: Store)(implicit ec: ExecutionContext) {
+  def validate(slot: Slot): Future[Validation] = {
```

Comment:
> Thinking about location for this method, what about on a companion object of Slot itself?

Sounds good! :+1: 

> This method is performing validation but is also side-effecty

You mean because `store` is mutable? That's definitely a something I agree isn't ideal here, particularly like you say because of the extra `Future` complexity. I was trying to capture something which would work generically for other validations we might do in future, and the `store` is the common component for all of them I think. But agreed let's try switching to just passing in the variants and see how we go.